### PR TITLE
Use CUB by default

### PR DESCRIPTION
**Pull request by asi1024**
https://github.com/cupy/cupy/issues/4897#issuecomment-1067850147

### CODE REVIEW

**Comment by kmaehashi** [comment]
/test mini

* Will you do docs in separate PR or in this PR?
* We will need to add test variant for `CUPY_ACCELERATOR=""` case.

File: cupy/_core/_accelerator.pyx [hunk 1]
```diff
@@ -43,7 +43,7 @@ def get_routine_accelerators():
 
 
 cdef _set_default_accelerators():
-    cdef str b, accelerator_names = os.getenv('CUPY_ACCELERATORS', '')
+    cdef str b, accelerator_names = os.getenv('CUPY_ACCELERATORS', 'cub')
```
Comment (asi1024) [inline cupy/_core/_accelerator.pyx:46]:
```suggestion
    cdef str b, accelerator_names = os.getenv('CUPY_ACCELERATORS', '' if runtime.is_hip else 'cub')
```

**Comment by asi1024** [comment]
@kmaehashi Updated. PTAL!

File: docs/source/reference/environment.rst [hunk 2]
```diff
@@ -71,7 +71,7 @@ Here are the environment variables that CuPy uses at runtime.
 
 .. envvar:: CUPY_ACCELERATORS
 
-  Default: ``""`` (no accelerators)
+  Default: ``"cub"`` (In ROCm HIP environment, the default value is ``""``. i.e., no accelerator is used.)
```
Comment (kmaehashi) [inline docs/source/reference/environment.rst:74]:
```suggestion
  Default: ``"cub"`` (In ROCm HIP environment, the default value is ``""``. i.e., no accelerators are used.)
```

File: docs/source/user_guide/performance.rst [hunk 3]
```diff
@@ -106,6 +106,9 @@ contiguous axes.
 In any case, we recommend to perform some benchmarks to determine whether CUB/cuTENSOR offers
 better performance or not.
 
+.. note::
+   CuPy v11 and above uses CUB by default. If you sets ``CUPY_ACCELERATORS=""`` it will be disabled.
```
Comment (kmaehashi) [inline docs/source/user_guide/performance.rst:110]:
```suggestion
   CuPy v11 and above uses CUB by default. To turn it off, you need to explicitly specify the environment variable ``CUPY_ACCELERATORS=""``.
```

**Review by kmaehashi — COMMENTED** [review]
Thanks, made tiny docs suggestions. Could you fix doctest failure?

**Comment by kmaehashi** [comment]
> We will need to add test variant for CUPY_ACCELERATOR="" case.

Did it in: https://github.com/cupy/cupy/pull/6580

**Comment by kmaehashi** [comment]
> Could you fix doctest failure? https://ci.preferred.jp/r/job/97008

* Add that this difference does not apply when CUB is enabled (CuPy v11+)
  https://docs.cupy.dev/en/latest/user_guide/difference.html#nan-not-a-number-handling
* Use `env:CUPY_ACCELERATORS: ""` instead of `env:CUPY_ACCELERATORS: null` so that doctests run with CUB disabled: https://github.com/cupy/cupy/blob/42718a96a1de4c9d1007624add585eda7239350f/.pfnci/matrix.yaml#L339-L362

**Review by kmaehashi — DISMISSED** [review]

**Comment by kmaehashi** [comment]
Ah wait, this is not done yet 😅 

> Add that this difference does not apply when CUB is enabled (CuPy v11+)
> https://docs.cupy.dev/en/latest/user_guide/difference.html#nan-not-a-number-handling

**Review by leofang — APPROVED** [review]
LGTM!

File: docs/source/user_guide/difference.rst [hunk 4]
```diff
@@ -178,3 +178,4 @@ counterparts:
 
 The reason is that internally the reduction is performed in a strided fashion, thus it does not ensure a proper
 comparison order and cannot follow NumPy's rule to always propagate the first-encountered NaN.
+Note that this difference does not apply when CUB is enabled (CuPy v11+)
```
Comment (kmaehashi) [inline docs/source/user_guide/difference.rst:181]:
```suggestion
Note that this difference does not apply when CUB is enabled (which is the default for CuPy v11 or later.)
```

**Pull request merged**